backtracking/graph_coloring.cpp

```diff
@@ -1,72 +1,106 @@
-#include <stdio.h>
+/**
+ * @file
+ * @brief prints the assigned colors
+ */
+#include <iostream>
+#include <array>
 
-// Number of vertices in the graph
-#define V 4
-
-void printSolution(int color[]);
+/**
+ * @namespace
+ * @brief Backtracking algorithms
+ */
+namespace backtracking {
+/** A utility function to print solution
+ * @param color description
+ * @param V number of vertices in the graph
+ */
+void printSolution(const int *color, int V) {
+    std::cout << "Following are the assigned colors\n";
+    for (int i = 0; i < V; i++) {
+        std::cout << color[i];
+    }
+    std::cout << "\n";
+}
 
-/* A utility function to check if the current color assignment
-   is safe for vertex v */
-bool isSafe(int v, bool graph[V][V], int color[], int c) {
-    for (int i = 0; i < V; i++)
-        if (graph[v][i] && c == color[i])
+/** A utility function to check if the current color assignment is safe for
+ * vertex v
+ * @tparam V number of vertices in the graph
+ * @param v description
+ * @param graph description
+ * @param color description
+ * @param c description
+ * @returns `true` if ....
+ * @returns `false` if ....
+ */
+template <int V>
+bool isSafe(int v, std::array <bool, V> graph, const int *color, int c) {
```

Comment:
please review the definitions for `std::array`.
This change implies that `graph` is a 1D array of type `bool` and length V. 
I think you mean to have a 2D array, i.e., an array of arrays:
```cpp
template <size_t V>
 bool isSafe(int v, const std::array<std::array <bool, V>> &graph, const int *color, int c) {
```